test/test_herder.py

```diff
@@ -157,12 +220,13 @@ def test_loop_detects_pidchange(self, open_mock, process_mock, sleep_mock):
     @patch('unicornherder.herder.time.sleep')
     @patch('unicornherder.herder.psutil.Process')
     @patch('%s.open' % builtin_mod)
-    def test_loop_reload_pidchange_signals(self, open_mock, process_mock,
-                                           sleep_mock):
+    def test_loop_reload_pidchange_signals(self, open_mock, process_mock, sleep_mock):
         proc1 = MagicMock()
         proc2 = MagicMock()
         proc1.pid = 123
         proc2.pid = 456
+        proc1.children = MagicMock(return_value=[proc2, "worker"])
+        proc2.children = MagicMock(return_value=["worker 1"])
```

Comment:
```suggestion
        proc1.children = Mock(return_value=[proc2, "worker"])
        proc2.children = Mock(return_value=["worker 1"])
```

I don't think these need to be `MagicMock`s.

Reply:
Hmm actually there's a lot of uses of `MagicMock` all across this code where it's not required... so it doesn't matter too much

Reply:
Sure, I had no idea that MagicMock and Mock were different things 🤷

Reply:
`MagicMock` is only relevant if you're needing to use so called "magic methods" which is broadly what other languages would call operator overloading.